doc(v1.7.0): update "Important Notes" (#951)
longhorn/website: content/docs/1.7.0/deploy/important-notes/index.md

Reviewer: The minimum XFS file system size behavior is new in v1.7.0. I added the note in https://github.com/longhorn/website/pull/905. IMO it should be in the important notes for this version. Is the omission accidental, or do you feel it should not be included?

Author: Yes, we need it. Sorry for mistakenly deleteing it. I added it back.

Reviewer: No worries!

- [General](#general)
  - [Supported Kubernetes Versions](#supported-kubernetes-versions)
  - [Pod Security Policies Disabled \& Pod Security Admission Introduction](#pod-security-policies-disabled--pod-security-admission-introduction)
  - [Introduction of Commandline Tool](#introduction-of-commandline-tool)
- [Networking](#networking)
  - [Storage Network Support for Read-Write-Many (RWX) Volumes](#storage-network-support-for-read-write-many-rwx-volumes)
- [Resilience](#resilience)
  - [RWX Volumes Fast Failover](#rwx-volumes-fast-failover)
  - [Support Configurable Timeout for Replica Rebuilding and Snapshot cloning](#support-configurable-timeout-for-replica-rebuilding-and-snapshot-cloning)
- [Data Integrity and Reliability](#data-integrity-and-reliability)
  - [Support Periodic and On-Demand Full Backups to Enhance Backup Reliability](#support-periodic-and-on-demand-full-backups-to-enhance-backup-reliability)
  - [High Availability of Backing Images](#high-availability-of-backing-images)
- [V2 Data Engine](#v2-data-engine)

>   - [Longhorn System Upgrade](#longhorn-system-upgrade)

Reviewer 2: Better _Longhorn System Live Upgrade_?

Author: No, it is not related to live upgrade. Before introducing the control plane of live upgrade, users need to detach volumes.

## V2 Data Engine

### Longhorn System Upgrade

Longhorn currently does not support live upgrading of V2 volumes. Ensure that all V2 volumes are detached before initiating the upgrade process.

### Introduction of Online Replica Rebuilding

Online replica rebuilding was introduced in Longhorn 1.7.0, so offline replica rebuilding has been removed.

### Block-type Disk Supports SPDK AIO, NVMe and VirtIO Bdev Drivers 

Before Longhorn v1.7.0, Longhorn block-type disks only supported the SPDK AIO bdev driver, which introduced extra performance penalties. Since v1.7.0, block devices can be directly managed by SPDK NVMe or VirtIO bdev drivers, improving IO performance through a kernel bypass scheme. For more information, see this [link](https://github.com/longhorn/longhorn/issues/7672).

### Introduction of Filesystem Trim

Filesystem trim is supported since Longhorn v1.7.0. If a disk is managed by the SPDK AIO bdev driver, the Trim (UNMAP) operation is not recommended in a production environment (ref). It is recommended to manage a block-type disk with an NVMe bdev driver.

### Linux Kernel on Longhorn Nodes

> Host machines with Linux kernel 5.15 may unexpectedly reboot when volume-related IO errors occur. To prevent this, update the Linux kernel on Longhorn nodes to version 5.19 or later. For more information, see [Prerequisites](../../v2-data-engine/prerequisites/). Additionally, version 6.7 or later is highly recommended for improved system stability.

Reviewer 3: ```suggestion
Host machines with Linux kernel 5.15 may unexpectedly reboot when volume-related IO errors occur. To prevent this, update the Linux kernel on Longhorn nodes to version 5.19 or later. For more information, see [Prerequisites](../../v2-data-engine/prerequisites/). Version 6.7 or later is recommended for improved system stability.
```

> ### Snapshot Creation Times Change from Time to Time

Reviewer 3: I had difficulty understanding this and line 131. Hopefully, I have understood the original issue sufficiently to suggest rewording.

```suggestion
### Snapshot Creation Time as shown in the UI occasionally changes
```

> Snapshots created before Longhorn v1.7.0 changes from Time to Time. This issue arises because the engine randomly selects a replica and its snapshot map each time the UI requests snapshot info, leading to potential 1-second gaps between snapshots among different replicas. This bug was fixed in v1.7.0, but snapshots created before this version may still encounter the issue. For more information, see this [link](https://github.com/longhorn/longhorn/issues/7641).

Reviewer 3: ```suggestion
Snapshots created before Longhorn v1.7.0 have their Creation Time, as shown in the UI, change occasionally. This issue arises because the engine randomly selects a replica and its snapshot map each time the UI requests snapshot info, leading to potential 1-second gaps between snapshots among different replicas. This bug was fixed in v1.7.0, but snapshots created before this version may still encounter the issue. For more information, see this [link](https://github.com/longhorn/longhorn/issues/7641).
```

> Snapshots created before Longhorn v1.7.0 changes from Time to Time. This issue arises because the engine randomly selects a replica and its snapshot map each time the UI requests snapshot info, leading to potential 1-second gaps between snapshots among different replicas. This bug was fixed in v1.7.0, but snapshots created before this version may still encounter the issue. For more information, see this [link](https://github.com/longhorn/longhorn/issues/7641).

Reviewer 2: Also when a volume rebuild is triggered the snapshot creation time changes

Author: Updated. Thank you.

> Snapshots created before Longhorn v1.7.0 may change occasionally. This issue arises because the engine randomly selects a replica and its snapshot map each time the UI requests snapshot information or when a replica is rebuilt with a random healthy replica. This can lead to potential 1-second gaps between snapshots among different replicas. Although this bug was fixed in v1.7.0, snapshots created before this version may still encounter the issue. For more information, see this [link](https://github.com/longhorn/longhorn/issues/7641).

Reviewer 2: _This can lead to potential 1-second gaps between snapshots among different replicas_
When the snapshot is rebuilt this is not true.

Author: Maybe`potential time gaps`?

### Unable to Revert a Volume to a Snapshot Created before Longhorn v1.7.0

> Reverting a volume to a snapshot created before Longhorn v1.7.0 is not supported due to an incorrect UserCreated flag set on the snapshot. A workaround is to back up the existing snapshots before upgrading to Longhorn v1.7.0 and restore them if needed. The bug is fixed in v1.7.0, and more information can be found [here](https://github.com/longhorn/longhorn/issues/9054).

Reviewer 3: A minor suggestion 'A workaround' suggests there may be more than one. Using 'The workaround' sounds more specific.

```suggestion
Reverting a volume to a snapshot created before Longhorn v1.7.0 is not supported due to an incorrect UserCreated flag set on the snapshot. The workaround is to back up the existing snapshots before upgrading to Longhorn v1.7.0 and restore them if needed. The bug is fixed in v1.7.0, and more information can be found [here](https://github.com/longhorn/longhorn/issues/9054).
```
